src/uct/api/uct.h

```diff
@@ -1432,23 +1448,52 @@ struct uct_listener_params {
     struct {
-        uint64_t             max_alloc; /**< Maximal allocation size */
-        size_t               max_reg;   /**< Maximal registration size */
-        uint64_t             flags;     /**< UCT_MD_FLAG_xx */
-        uint64_t             reg_mem_types; /**< Bitmap of memory types that Memory Domain can be registered with */
-        uint64_t             cache_mem_types; /**< Bitmap of memory types that can be cached for this Memory Domain */
-        uint64_t             detect_mem_types; /**< Bitmap of memory types that Memory Domain can detect if address belongs to it */
-        uint64_t             alloc_mem_types;  /**< Bitmap of memory types that Memory Domain can allocate memory on */
-        uint64_t             access_mem_types; /**< Memory types that Memory Domain can access */
+        /** Maximal allocation size */
+        uint64_t max_alloc;
+
+        /** Maximal registration size */
+        size_t   max_reg;
+
+        /** Memory Domain capability flags with UCT_MD_FLAG_xx format */
+        uint64_t flags;
+
+        /** Bitmap of memory types that Memory Domain can be registered with */
+        uint64_t reg_mem_types;
+
+        /**< Bitmap of memory types that can be cached for this Memory Domain */
+        uint64_t cache_mem_types;
+
+        /** Bitmap of memory types that Memory Domain can detect if address
+         *  belongs to it */
+        uint64_t detect_mem_types;
+
+        /** Bitmap of memory types that Memory Domain can allocate memory on */
+        uint64_t alloc_mem_types;
+
+        /** Memory types that Memory Domain can access */
+        uint64_t access_mem_types;
     } cap;
 
-    ucs_linear_func_t        reg_cost;  /**< Memory registration cost estimation
-                                             (time,seconds) as a linear function
-                                             of the buffer size. */
+    /** Memory registration cost estimation (time,seconds) as a linear function
+     *  of the buffer size */
+    ucs_linear_func_t reg_cost;
+
+    /** Component name */
+    char              component_name[UCT_COMPONENT_NAME_MAX];
+
+    /** Size of buffer needed for packed rkey */
+    size_t            rkey_packed_size;
+
+    /** Mask of CPUs near the resource */
+    ucs_cpu_set_t     local_cpus;
+
+    /** Size of buffer needed for packed shared mkey. Valid only if
+     *  @ref UCT_MD_FLAG_SHARED_MKEY is supported by Memory Domain */
+    size_t            shared_mkey_packed_size;
 
-    char                     component_name[UCT_COMPONENT_NAME_MAX]; /**< Component name */
-    size_t                   rkey_packed_size; /**< Size of buffer needed for packed rkey */
-    ucs_cpu_set_t            local_cpus;    /**< Mask of CPUs near the resource */
+    /** Global id - unique device identifier in a system (can be MAC address or
+     *  GUID), '\0' means that an ID is unknown and should not be used */
+    char              global_id[UCT_MD_GLOBAL_ID_MAX];
```

Comment:
this breaks backward compatibility: application that was compiled with older version will get memory overwrite because its struct is smaller
need to depend on https://github.com/openucx/ucx/pull/8428